src/nfts/ERC1155AdminMinter.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.18;
+
+import {Roles} from "@protocol/core/Roles.sol";
+import {CoreRef} from "@protocol/refs/CoreRef.sol";
+import {ERC1155MaxSupplyMintable} from "@protocol/nfts/ERC1155MaxSupplyMintable.sol";
+
+contract ERC1155AdminMinter is CoreRef {
+    /// --------- Structs ---------- ///
+
+    struct BulkMint {
+        address nftContract;
+        uint256 tokenId;
+        uint256 amount;
+        address recipient;
+    }
+
+    /// --------- Events ---------- ///
+
+    /// @notice - Event emitted when the admin mint is successful
+    event TokensMinted(
+        address indexed nftContract,
+        address indexed recipient,
+        uint256 amount,
+        uint256 tokenId
+    );
+
+    constructor(address _core) CoreRef(_core) {}
```

Comment:
What about adding whitelisting support, so we can limit this to specific contract/s?